src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -985,6 +986,7 @@ private void InitODataParser()
                 {
                     IsNullable = (bool)columnInfoFromAdapter["AllowDBNull"],
                     IsAutoGenerated = (bool)columnInfoFromAdapter["IsAutoIncrement"],
+                    IsReadOnly = (bool)columnInfoFromAdapter["IsAutoIncrement"],
```

Comment:
IsAutoIncrement is just a special case of read only. Computed/timestamp being other ones. There is no specific mapping for read-only columns but if the column is auto-gen, then we can straight away start with IsReadOnly  = IsAutoIncrement = true.

Reply:
https://github.com/Azure/data-api-builder/pull/1596#discussion_r1308230767

Reply:
> A computed column or a timestamp/rowversion column is not autogenerated but is read only.

Such a distiction is artificially made by DAB.. right? SQL Server metadata will be provided by `columnInfoFromAdapter["IsAutoIncrement"],` which is used for both `IsReadOnly` and `IsAutoGenerated`. I don't see how there could ever be a circumstance where `IsReadOnly` and `IsAutoGenerated` will be different.

Reply:
Perhaps a comment in the code to note that the `IsReadOnly` field will be overridden later by `PopulateColumnDefinitionsWithReadOnlyFlag()` and is only set here because `IsReadOnly` is not nullable.

Reply:
added